### PR TITLE
Map model library locally

### PR DESCRIPTION
**Pull request by CharlieFRuan**
Instead of relying on `local_id` and `model_lib` in the `mlc-chat-config.json` on huggingface, we map it internally within AppConfig by adding a field `model_lib` in each `ModelRecord` in `model_list`.

### CODE REVIEW

**Comment by CharlieFRuan** [comment]
cc @tqchen

File: README.md [hunk 1]
```diff
@@ -131,6 +131,7 @@ async main() {
     "model_list": [
       {
         "model_url": myLlamaUrl,
+        "model_lib": "llama-v1-3b-q4f32_0",
```
Comment (tqchen) [inline README.md:134]: we can use model_lib: to directly point to a wasm url, so remove the model_lib indirection

Comment (tqchen) [reply README.md:134]: That also enables us to remove the lib map indirection and focus on concept of "model" and "model library"

**Comment by CharlieFRuan** [comment]
@tqchen Just removed `model_lib_map`. I am still keeping the interface `AppConfig` despite it now only having one required field `model_list` after removing the map; was thinking that it could be useful for cases like `simple-chat`'s `"use_web_worker"`. Please let me know if you think we should remove `AppConfig` altogether.

Besides, should we also rename `local_id` to perhaps `model_name`? Though the name is still used in mlc-llm.

**Comment by tqchen** [comment]
Great, let us think about the name `local_id` and see what to do

**Pull request merged**